port/aix/omrsl.c

```diff
@@ -763,5 +763,80 @@ omrsl_startup(struct OMRPortLibrary *portLibrary)
 uintptr_t
 omrsl_get_libraries(struct OMRPortLibrary *portLibrary, OMRLibraryInfoCallback callback, void *userData)
 {
-	return OMRPORT_ERROR_NOT_SUPPORTED_ON_THIS_PLATFORM;
+	uintptr_t result = 0;
+	void *buffer = NULL;
+	struct ld_info *ldInfo = NULL;
+	int rc = 0;
+	unsigned int bufferSize = 65536;
+	for (;;) {
+		buffer = portLibrary->mem_allocate_memory(
+				portLibrary,
+				bufferSize,
+				OMR_GET_CALLSITE(),
+				OMRMEM_CATEGORY_PORT_LIBRARY);
+		if (NULL == buffer) {
+			portLibrary->error_set_last_error_with_message(
+					portLibrary,
+					-1,
+					"Failed to allocate memory for loadquery buffer.");
+			return (uintptr_t)(intptr_t)-1;
+		}
+		rc = loadquery(L_GETINFO, buffer, bufferSize);
+		if (-1 != rc) {
+			break;
+		}
+		if (ENOMEM != errno) {
+			int32_t portableError = portLibrary->error_last_error_number(portLibrary);
+			portLibrary->error_set_last_error_with_message(
+					portLibrary,
+					portableError,
+					"loadquery(L_GETINFO) failed.");
+			portLibrary->mem_free_memory(portLibrary, buffer);
+			return (uintptr_t)(intptr_t)portableError;
+		}
+		portLibrary->mem_free_memory(portLibrary, buffer);
+		bufferSize *= 2;
+	}
+	ldInfo = (struct ld_info *) buffer;
+	for(;;) {
+		if ('\0' == ldInfo->ldinfo_filename[0]) {
+			goto next_entry;
+			continue;
+		}
+		const char* pathName = ldInfo->ldinfo_filename;
+		const char* memberName = pathName + strlen(pathName) + 1;
+		char fullLibName[PATH_MAX];
+		if ('\0' != memberName[0]) {
+			snprintf(fullLibName, sizeof(fullLibName), "%s(%s)", pathName, memberName);
+		}
+		else {
+			strncpy(fullLibName, pathName, sizeof(fullLibName));
+			fullLibName[sizeof(fullLibName) - 1] = '\0';
+		}
```

Comment:
There's no need to copy the path if there's no member name; just use `pathName` which points to `fullLibName` when there is a member name.